cpp/src/arrow/python/type_traits.h

```diff
@@ -127,8 +134,14 @@ template <>
 struct npy_traits<NPY_OBJECT> {
   typedef PyObject* value_type;
   static constexpr bool supports_nulls = true;
+
+  static inline bool isnull(PyObject* v) { return v != Py_None; }
```

Comment:
`inline` is redundant here: http://en.cppreference.com/w/cpp/language/inline.

> A function defined entirely inside a class/struct/union definition, whether it's a member function or a non-member friend function, is implicitly an inline function.

Reply:
I see. This is really using the same convention as the rest of the file, though.

Reply:
Hm, so that's also called `isnull`. Shouldn't that mean `v == Py_None`?

Reply:
Probably needs a test as well since it isn't failing.

Reply:
Nice catch :-) I'm not sure how to test it. Defining `isnull` is necessary for compiling, but that path isn't taken at runtime as object arrays are handled separately.